src/workerd/api/pyodide/pyodide.h

```diff
@@ -229,6 +262,16 @@ class ArtifactBundler: public jsg::Object {
     return false;  // TODO(later): Remove this function once we regenerate the bundle.
   }
 
+  kj::Maybe<jsg::Ref<PackagesTarReader>> getPackage(kj::String path) {
+    KJ_IF_SOME(pacman, packageManager) {
+      KJ_IF_SOME(ptr, pacman.getPyodidePackage(path)) {
+        return jsg::alloc<PackagesTarReader>(ptr);
+      }
+    }
+
+    return kj::none;
```

Comment:
Is it a logic error if we reach this point?

Reply:
Yes, it is a logic error, but it ends up getting surfaced by the JavaScript caller instead of the C++ code. I think it's more flexible for the API to return a nullable value here and make it the caller's responsibility to check. There's also not a huge risk for missing a nullability check since the TypeScript type matches.